### PR TITLE
Clear the meta data dev before creating thin pool

### PR DESCRIPTION
**Pull request by trgill**
If the given device is re-used stale data may cause problems with DM.

Signed-off-by: Todd Gill <tgill@redhat.com>

### CODE REVIEW

**Comment by trgill** [comment]
The check looks to have failed due known issue with loopback devices.

File: tests/util/simple_tests.rs [hunk 1]
```diff
@@ -156,6 +156,17 @@ pub fn test_thinpool_device(paths: &[&Path]) -> () {
                                       &dm,
                                       vec![metadata_blockdev.avail_range()])
             .unwrap();
+
+    // Clear the first MiB of the meta data dev.  If the given device is bing re-used
```
Comment (mulkieran) [inline tests/util/simple_tests.rs]: Might as well fix typo...

File: tests/util/simple_tests.rs [hunk 2]
```diff
@@ -156,6 +156,17 @@ pub fn test_thinpool_device(paths: &[&Path]) -> () {
...
+    // stale data may cause problems with DM.
+    wipe_sectors(&metadata_dev
+                      .devnode()
+                      .expect("metadata_dev must have devnode"),
```
Comment (mulkieran) [inline tests/util/simple_tests.rs]: just unwrap() here, it's a test.

File: tests/util/simple_tests.rs [hunk 3]
```diff
@@ -156,6 +156,17 @@ pub fn test_thinpool_device(paths: &[&Path]) -> () {
                                       &dm,
                                       vec![metadata_blockdev.avail_range()])
             .unwrap();
+
+    // Clear the first MiB of the meta data dev.  If the given device is bing re-used
+    // stale data may cause problems with DM.
```
Comment (mulkieran) [inline tests/util/simple_tests.rs]: If possible, could you be more specific about what the problem is?

Comment (trgill) [inline tests/util/simple_tests.rs]: Ok

File: tests/util/simple_tests.rs (hunk 2 above)
Comment (trgill) [inline tests/util/simple_tests.rs]: Ok

File: tests/util/simple_tests.rs [hunk 4]
```diff
@@ -156,6 +156,16 @@ pub fn test_thinpool_device(paths: &[&Path]) -> () {
                                       &dm,
                                       vec![metadata_blockdev.avail_range()])
             .unwrap();
+
+    // Clear the first MiB of the meta data dev.  DM will look at the first block
+    // on the meta data dev.  If the first block is not all zeros - it will cause
+    // the device to apear as an existing meta rather than a new one.
```
Comment (mulkieran) [inline tests/util/simple_tests.rs]: Please fix typo.

File: tests/util/simple_tests.rs [hunk 5]
```diff
@@ -156,6 +156,16 @@ pub fn test_thinpool_device(paths: &[&Path]) -> () {
                                       &dm,
                                       vec![metadata_blockdev.avail_range()])
             .unwrap();
+
+    // Clear the first MiB of the meta data dev.  DM will look at the first block
+    // on the meta data dev.  If the first block is not all zeros - it will cause
+    // the device to appear as an existing meta rather than a new one.
+    wipe_sectors(&metadata_dev.devnode().unwrap(),
+                 Sectors(0),
+                 Bytes(IEC::Mi as u64).sectors())
```
Comment (mulkieran) [inline tests/util/simple_tests.rs]: What is a block? Is it a sector, should you just be using ```Sectors(1)```?

**Comment by trgill** [comment]
As discussed, I updated to clear the entire device.

**Comment by agrover** [comment]
tests just need fixing for latest dm-rs API change.

File: tests/util/simple_tests.rs [hunk 6]
```diff
@@ -25,6 +25,7 @@ use self::tempdir::TempDir;
 use self::uuid::Uuid;
 
 use libstratis::engine::{Engine, EngineError, ErrorEnum};
+use libstratis::engine::strat_engine::device::wipe_sectors;
```
Comment (mulkieran) [inline tests/util/simple_tests.rs]: Put this import with the other ```device``` imports.

Comment (mulkieran) [inline tests/util/simple_tests.rs]: Fix the import and rebase on master and I expect everything will be good to go.

**Pull request merged**